packages/devtools_app/test/cpu_profiler_controller_test.dart

```diff
@@ -109,6 +110,68 @@ void main() {
       );
     });
 
+    test('processDataForTag', () async {
+      final cpuProfileDataWithTags =
+          CpuProfileData.parse(cpuProfileDataWithUserTagsJson);
+      await controller.transformer.processData(cpuProfileDataWithTags);
+      controller.loadProcessedData(cpuProfileDataWithTags);
+
+      expect(
+          controller.dataNotifier.value.cpuProfileRoot.profileMetaData.time
+              .duration.inMicroseconds,
+          equals(250));
+      expect(
+          controller.dataNotifier.value.cpuProfileRoot.toStringDeep(),
+          equals(
+            '''
+  all - children: 1 - excl: 0 - incl: 5
+    Frame1 - children: 2 - excl: 0 - incl: 5
+      Frame2 - children: 2 - excl: 0 - incl: 2
+        Frame3 - children: 0 - excl: 1 - incl: 1
+        Frame4 - children: 0 - excl: 1 - incl: 1
+      Frame5 - children: 1 - excl: 2 - incl: 3
+        Frame6 - children: 0 - excl: 1 - incl: 1
+''',
+          ));
+
+      await controller.loadDataWithTag('userTagA');
+      expect(
+          controller.dataNotifier.value.cpuProfileRoot.toStringDeep(),
+          equals(
+            '''
+  all - children: 1 - excl: 0 - incl: 2
+    Frame1 - children: 2 - excl: 0 - incl: 2
+      Frame2 - children: 1 - excl: 0 - incl: 1
+        Frame3 - children: 0 - excl: 1 - incl: 1
+      Frame5 - children: 0 - excl: 1 - incl: 1
+''',
+          ));
+
+      await controller.loadDataWithTag('userTagB');
+      expect(
+          controller.dataNotifier.value.cpuProfileRoot.toStringDeep(),
+          equals(
+            '''
+  all - children: 1 - excl: 0 - incl: 1
+    Frame1 - children: 1 - excl: 0 - incl: 1
+      Frame2 - children: 1 - excl: 0 - incl: 1
+        Frame4 - children: 0 - excl: 1 - incl: 1
+''',
+          ));
```

Comment:
nit: consider some trailing commas to force flutter style.

Reply:
done